packages/block-editor/src/utils/dom.js

```diff
@@ -156,11 +154,24 @@ export function getVisibleElementBounds( element ) {
 		for ( const child of currentElement.children ) {
 			if ( isElementVisible( child ) ) {
 				const childBounds = child.getBoundingClientRect();
-				bounds = rectUnion( bounds, childBounds, viewportRect );
+				bounds = rectUnion( bounds, childBounds );
 				stack.push( child );
 			}
 		}
 	}
 
+	/*
+	 * Take into account the outer horizontal limits of the container in which
+	 * an element is supposed to be "visible". For example, if an element is
+	 * positioned -10px to the left of the window x value (0), this function
+	 * discounts the negative overhang because it's not visible and therefore
+	 * not to be counted in the visibility calculations. Top and bottom values
+	 * are not accounted for to accommodate vertical scroll.
+	 */
+	bounds = rectIntersect(
+		bounds,
+		new window.DOMRect( 0, -Infinity, viewport.innerWidth, Infinity )
```

Comment:
Optional, if it's immutable then `DOMRectReadOnly` is 👍🏻 

```suggestion
		new window.DOMRectReadOnly( 0, -Infinity, viewport.innerWidth, Infinity )
```